### ISSUE TITLE
SafeAreaView flow type error in YellowBox

### ISSUE THREAD

**Issue by jjenzz** [facebook/react-native#21897]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment

```
  React Native Environment Info:
    System:
      OS: macOS 10.14
      CPU: x64 Intel(R) Core(TM) i7-6567U CPU @ 3.30GHz
      Memory: 2.18 GB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.3.0 - ~/.nvm/versions/node/v10.3.0/bin/node
      Yarn: 1.6.0 - ~/.yarn/bin/yarn
      npm: 6.1.0 - ~/.nvm/versions/node/v10.3.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.0, macOS 10.14, tvOS 12.0, watchOS 5.0
    IDEs:
      Android Studio: 3.1 AI-173.4819257
      Xcode: 10.0/10A255 - /usr/bin/xcodebuild
    npmPackages:
      @storybook/react-native: ^3.4.11 => 3.4.11
      react: 16.5.0 => 16.5.0
      react-native: 0.57.2 => 0.57.2
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

## Description

![image](https://user-images.githubusercontent.com/175330/47306305-f22c7780-d623-11e8-9ee0-321db325aa56.png)

![image](https://user-images.githubusercontent.com/175330/47306313-f8225880-d623-11e8-8b83-9acf69a8f900.png)

## Reproducible Demo
1. Initialise new react native project with flow
2. `yarn flow check`
3. The [following errors appear](https://github.com/facebook/react-native/issues/21898):
    ![image](https://user-images.githubusercontent.com/175330/47306419-4d5e6a00-d624-11e8-85c9-7c7c28ab6c24.png)
4. Add those files to `[untyped]` config:

    ```
    [untyped]
    .*/node_modules/react-native/Libraries/Components/Navigation/NavigatorIOS.ios.js
    .*/node_modules/react-native/Libraries/Components/TextInput/InputAccessoryView.js
    .*/node_modules/react-native/Libraries/Components/View/ViewPropTypes.js
    ```
5. `yarn flow check` again
6. YellowBox errors appear

**Comment by jjenzz** [comment]
@react-native-bot env info says `0.57.2` 🙈 😛 

Please remove `Old version` label.

**Comment by radko93** [comment]
@jjenzz latest is 0.57.3. Maybe consider submitting a PR to fix this issues?

**Comment by jjenzz** [comment]
@radko blimey, seems I can’t keep up haha. I’ll re-test in 0.53.3. I actually raised this with the intention to fix myself but haven’t gotten around to it yet 🙈

**Comment by radko93** [comment]
Latest is 0.57.4 now :)

**Comment by nate-whil** [comment]
The issue still exists with the files specified by @jjenzz  with `"react-native": "0.57.8"` when upgrading from `"flow-bin": "0.81.0"` to `"flow-bin": "0.82.0"`

**Comment by thymikee** [comment]
Hey! We can't actually see the reported bug on latest RN 0.59 so we'll close this issue, as this was likely fixed in 0.58 or later.
